### PR TITLE
HPCC-17169 Silence/fix lint warnings

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes HPCC-17169

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please Review

Note if you enable travis, it will auto lint on push to github like this:  https://travis-ci.org/GordonSmith/HPCC-Platform/builds/207435610

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo : false
 language: node_js
 node_js:
-  - "4.2"
+  - "6.9.1"
```
Comment (miguelvazq) [inline .travis.yml]: Are you requiring this _exact_ version of node for this? Not sure if `>=6.9.1` works in this syntax

Comment (GordonSmith) [inline .travis.yml]: It dictates which version of node Travis will use to execute the linting engine.  It has nothing to do with our JS or our build environment.

Comment (miguelvazq) [inline .travis.yml]: Looks good then.

**Comment by richardkchapman** [comment]
@GordonSmith One question

File: esp/src/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,1213 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (GordonSmith) [inline esp/src/yarn.lock]: Yes - it ensures that all dependencies of dependencies are "locked" at a known versions - even if their semantic version dependencies allow for patch / minor updates.

IOW it ensure that if we gran all the dependencies in the future, they will match.

**Comment by GordonSmith** [comment]
@richardkchapman answered (and yes)

**Pull request merged**